docs/specification/2025-03-26/basic/authorization.md

```diff
@@ -4,242 +4,394 @@ type: docs
+
+MCP servers that do not implement a [metadata discovery process](#23-server-metadata-discovery) must
+be assumed to implement the OAuth flow through standard endpoints (e.g., `/authorize` and `/token`)
+
+#### 2.4.2 Flow Description
+
+The third-party authorization flow comprises these steps:
+
+1. MCP client requests data from the MCP server.  
+1. MCP server responds with *HTTP 401 Unauthorized* and `WWW-Authenticate` header containing the
+   `issuer`, `scope`, and `resource`.  
+1. MCP client retrieves the `/authorize` endpoint from the `issuer` metadata document.  
+1. MCP client redirects user to identity provider's `/authorize` endpoint. MCP client uses `scope` and
+   `resource` parameters provided in the `WWW-Authenticate` header in the `/authorize` URL.  
+1. User authorizes with identity provider with `scope` and `resource`.  
+1. Identity provider redirects back to MCP client with authorization code.  
+1. MCP client exchanges the code for the requested tokens.  
+1. MCP client caches the token.  
+1. MCP clients can now request data from the MCP server with the token attached in the `Authorization`
+   header.
+
+```mermaid
+sequenceDiagram
+    participant UA as User-Agent (Browser)
+    participant MC as MCP Client
+    participant MS as MCP Server
+    participant IDP as Third-Party Auth Server
+
+    MC->>MS: Request data
+    MS-->>MC: 401 Unauthorized with WWW-Authenticate
+    Note over MC: MCP Client retrieves /authorize endpoint URI from the issuer metadata document (Section 2.3)
+    MC-->>UA: Initiate authorization request at /authorize
+    Note over UA: User authenticates and authorizes client for scope and resource
+    UA->>MC: Return auth code to client
+    MC->>IDP: Exchange code for token
+    IDP-->>MC: Third-party access token
+    MC->>MC: Cache token
+```
+
+#### 2.4.3 Session Binding Requirements
+
+MCP servers integrating with third-party authorization services **MUST**:
+
+1. Maintain secure mapping between third-party tokens and issued client sessions.  
+2. Validate third-party token status before honoring MCP tokens.  
+3. Implement appropriate token lifecycle management.  
+4. Provide appropriate responses to the client if re-authentication is needed. Please refer to
+   [RFC 9470 OAuth 2.0 Step Up Authentication Challenge Protocol](https://www.rfc-editor.org/rfc/rfc9470.html).
+
+#### 2.4.4 Security Considerations
+
+When implementing authorization with identity providers, MCP servers **MUST**:
+
+1. Request the minimum number of required scopes.  
+2. Securely store any credential artifacts obtained from the client and/or the IdP.  
+3. Implement appropriate session timeout handling.  
+4. Consider security implications of token chaining.  
+5. Implement proper error handling for third-party auth failures.
+
+MCP clients **MUST**:
+
+1. Securely store authentication tokens received from the identity provider.  
```

Comment:
Perhaps be more explicit about what is included as "authentication tokens",. For example can we use the OAuth terminology - Access Tokens, Refresh Tokens to avoid confusion?